save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -54,4 +70,24 @@ internal class ContestController(
     ): Flux<ContestDto> = Flux.fromIterable(
         contestService.findFinishedContests(pageSize)
     ).map { it.toDto() }
+
+    /**
+     * @param contestName
+     * @return Organization
+     */
+    @GetMapping("/{contestName}/public-test")
+    @PreAuthorize("permitAll()")
+    @Suppress("UnsafeCallOnNullableType")
+    fun getPublicTestForContest(
+        @PathVariable contestName: String,
+    ): Mono<PublicTestDto> {
+        val contest = contestService.findByName(contestName).get()
+        val testSuite = testSuitesService.findTestSuiteById(
+            contest.testSuiteIds?.split(",")?.first()?.toLong()!!
+        ).get()
+        val test = testService.findTestsByTestSuiteId(testSuite.id!!).first()
```

Comment:
While exception possibly produced by `Optional#get()` is more or less self-descriptive, when you'll get `Collection is empty` from here it won't be as descriptive. Please change to `firstOrNull` and handle missing value